ctapipe/image/modifications.py

```diff
@@ -0,0 +1,113 @@
+from abc import abstractmethod
+import numpy as np
+from ..core.component import TelescopeComponent
+from ..core.traits import FloatTelescopeParameter, BoolTelescopeParameter
+
+
+def add_noise(image, noise_level, rng=None, correct_bias=True):
+    """
+    Create a new image with added poissonian noise
+    """
+    if not rng:
+        rng = np.random.default_rng()
+    noisy_image = image.copy()
+    noise = rng.poisson(noise_level)
+    noisy_image += noise
+    if correct_bias:
+        noisy_image -= noise_level
+    return noisy_image
+
+
+def smear_image(image, geom, smear_factor):
+    """
+    Create a new image with values smeared to the direct pixel neighbors
+    Pixels at the camera edge lose charge this way.
+    """
+    # make clear what smear factor is supposed to mean and that only direct neighbors are taken into account
+    # smear factor can be an array as well -> selection in tool possible!
+    # a more sophisticated approach might make use of the pixel area, but thats complicated
+    if geom.pix_type.value == "hexagon":
+        max_neighbors = 6
+    elif geom.pix_type.value == "rectangular":  # thats probably labeld differently
+        max_neighbors = 8  # or 4? lookup neighbor matrix for a rect cam
+    else:
+        print(geom.pix_type)
+        raise Exception("Unknown pixel type")
+
+    diffused_image = (
+        (image * geom.neighbor_matrix).sum(axis=1) * smear_factor / max_neighbors
+    )
+    remaining_image = image * (1 - smear_factor)
+    smeared_image = remaining_image + diffused_image
+    return smeared_image
+
+
+class ImageModifier(TelescopeComponent):
+    """
+    Abstract class for configurable image modifying algorithms. Use
+    ``ImageModifier.from_name()`` to construct an instance of a particular algorithm
+    """
+
+    @abstractmethod
+    def __call__(self, tel_id: int, image: np.ndarray) -> np.ndarray:
+        """
+        Modifies a given image
+
+        Returns
+        -------
+        np.ndarray
+            modified image
+        """
+        pass
+
+
+class NullModifier(ImageModifier):
+    def __call__(self, tel_id: int, image: np.ndarray) -> np.ndarray:
+        return image
+
+
+class LSTImageModifier(ImageModifier):
+    """
+    Add in everything lstchain does.
+    """
+
+    smear_factor = FloatTelescopeParameter(
+        default_value=0.2, help="Fraction of light to move to each neighbor"
+    ).tag(config=True)
+    transition_charge = FloatTelescopeParameter(
+        default_value=8, help="separation between dim and bright pixels"
+    ).tag(config=True)
+    dim_pixel_bias = FloatTelescopeParameter(
+        default_value=0.6, help="extra bias to add in dim pixels"
+    ).tag(config=True)
+    dim_pixel_noise = FloatTelescopeParameter(
+        default_value=1.5, help="expected extra noise in dim pixels"
+    ).tag(config=True)
+    bright_pixel_noise = FloatTelescopeParameter(
+        default_value=1.44, help="expected extra noise in bright pixels"
```

Comment:
Why this strange default? 

I would set the defaults all to 0 / 1, so that by default, the ImageModifier is a no-op

Reply:
Copied from the lstchain pr. But I'm fine with that

Reply:
Just curious: why is this used in the LST chain? Is it basically another calibration to remove some strange effects of the ring sampler?  Is that a calibration we should include somewhere earlier, e.g. where do the values get derived?

Reply:
Nothing to do with the ring sampler. It's just to make the simulations simulated for specific conditions (extragalactic dark night) fit better for other kinds of observations (e.g. galactiv fov for crab) and a worse PSF than simulated.

Reply:
@moralejo showed nice results of how these two additions improve the data mc mismatches quite dramatically.

Reply:
See discussion here: https://github.com/cta-observatory/cta-lstchain/pull/708

Reply:
ah, so it's to add NSB essentially - yes hat makes sense (though I don't quite understand the smearing or part or why dim pixels respond differently).  Would  be interesting to test how different adding the noise here is from adding it at the simulation level - that could save some time in prod5 to not simulate different NSB levels if the effect is the same

Reply:
It does sounds similar to what we used to do in Whipple: add noise to ON/OFF observations such that you bring the total NSB noise up to the level of the worst of the pair.

Reply:
> > though I don't quite understand the smearing or part or why dim pixels respond differentl
> 
> I think abelardos reasoning for this was that in noise dominated pixels there is a positive bias for peak-finding integrators, since they will look for a high peak, even if there is only noise. So noise is higher in pixels where you integrate over the highest fluctuation as opposed to just integrating over the pulse with the normal noise contribution. This is also the reason why you can give an additional `dim_bias`.

Yes, that is the idea: it is not possible to do a 100% correct tuning at the DL1a level if a peak-search algorithm was used. Only the limit case of bright signals, for which the integration window position won't change upon adding some extra noise, is easy to treat (I consider "bright signals" those above the picture threshold).  What I did in lstchain is an empirical tuning of the noise-only spectrum for interleaved pedestals, which is what I apply for pixels below the picture threshold. 

Bear in mind that the idea is to use this for fine-tuning, it should not replace having MC generated with different levels of NSB (either at the sim_telarray level, or by adding noise in R1 waveforms)